From 080b9d6552f150cce3e49ca1c88b1546eb785f19 Mon Sep 17 00:00:00 2001
From: George Cherian <george.cherian@marvell.com>
Date: Mon, 24 May 2021 12:55:10 +0530
Subject: [PATCH 15/23] octeontx2-af: Fix the KPU support to parse PPPoE
 packets

commit 83e524132e086f596699487236a14449b3478474  from
git@git.assembla.com:cavium/WindRiver.linux.git

The KPU support for the following
 1. L2 + PPPoE + PPP + IPv4
 2. L2 + PPPoE + PPP + IPv6

flows were dropped by the default mkex as the DMAC extraction
failed.

Fixes: ea11dadf0423("octeontx2-af: Add KPU support to parse PPPoE
packets")
Signed-off-by: George Cherian <george.cherian@marvell.com>

Change-Id: Iaed55d715c484e41480f23af9ef979787d8345d0
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/52480
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
index 354d9107f5d2..04f2bd7535d6 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
@@ -8729,7 +8729,7 @@ static struct npc_kpu_profile_action kpu1_action_entries[] = {
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
 		8, 12, 0, 2, 0,
 		NPC_S_KPU4_PPPOE, 12, 1,
-		NPC_LID_LB, NPC_LT_LB_PPPOE,
+		NPC_LID_LA, NPC_LT_LA_ETHER,
 		0,
 		0, 0, 0, 0,
 	},
-- 
2.31.1

